lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs

```diff
@@ -0,0 +1,173 @@
+import valueParser from 'postcss-value-parser';
+
+import { isPlainObject, isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { declarationValueIndex } from '../../utils/nodeFieldIndices.mjs';
+import getDeclarationValue from '../../utils/getDeclarationValue.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import setDeclarationValue from '../../utils/setDeclarationValue.mjs';
+import { singleValueColorProperties } from '../../reference/properties.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'declaration-property-value-keyword-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	expected: (actual, expected) => `Expected "${actual}" to be "${expected}"`,
+	rejected: (keyword, property) =>
+		`Unexpected deprecated keyword "${keyword}" for property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/declaration-property-value-keyword-no-deprecated',
+	fixable: true,
+};
+
+const DEPRECATED_COLORS = {
+	activecaption: 'canvas',
+	appworkspace: 'canvas',
+	background: 'canvas',
+	inactivecaption: 'canvas',
+	infobackground: 'canvas',
+	menu: 'canvas',
+	scrollbar: 'canvas',
+	window: 'canvas',
+	activeborder: 'ButtonBorder',
+	inactiveborder: 'ButtonBorder',
+	threeddarkshadow: 'ButtonBorder',
+	threedhighlight: 'ButtonBorder',
+	threedlightshadow: 'ButtonBorder',
+	threedshadow: 'ButtonBorder',
+	windowframe: 'ButtonBorder',
+	captiontext: 'CanvasText',
+	infotext: 'CanvasText',
+	menutext: 'CanvasText',
+	windowtext: 'CanvasText',
+	buttonhighlight: 'ButtonFace',
+	buttonshadow: 'ButtonFace',
+	threedface: 'ButtonFace',
+	inactivecaptiontext: 'GrayText',
+};
+const COLOR_PROPERTIES = [...singleValueColorProperties.values()].map(
+	(value) => /** @type {[string, Record<String, string>]} */ ([value, DEPRECATED_COLORS]),
+);
+const PROPERTY_NAME_TO_KEYWORD = new Map([
+	[
+		'appearance',
+		/** @type {string | string[] | Record<string, string>} */ ({
+			button: 'auto',
+			checkbox: 'auto',
+			listbox: 'auto',
+			menulist: 'auto',
+			meter: 'auto',
+			'progress-bar': 'auto',
+			'push-button': 'auto',
+			radio: 'auto',
+			searchfield: 'auto',
+			'slider-horizontal': 'auto',
+			'square-button': 'auto',
+			textarea: 'auto',
+		}),
+	],
+	[
+		'image-rendering',
+		{
+			optimizeQuality: 'smooth',
+			optimizeSpeed: 'pixelated',
+		},
+	],
+	['overflow', { overlay: 'auto' }],
+	['text-justify', { distribute: 'inter-character' }],
+	['text-orientation', { 'sideways-right': 'sideways' }],
+	['user-select', { element: 'contain' }],
+	['zoom', { reset: '1' }],
+	['text-decoration-line', 'blink'],
+	['box-sizing', 'padding-box'],
+	['image-orientation', 'flip'],
```

Comment:
```suggestion
	['text-decoration-line', ['blink']],
	['box-sizing', ['padding-box']],
	['image-orientation', ['flip']],
```

Can these be an array one value to simplify the `isArray` and `typeOf` conditionals?

Reply:
That's a good refactor because it enables us to use the plural later on.